src/coreclr/vm/comutilnative.cpp

```diff
@@ -1624,7 +1624,8 @@ BOOL CanCompareBitsOrUseFastGetHashCode(MethodTable* mt)
     }
 
     if (mt->ContainsPointers()
-        || mt->IsNotTightlyPacked())
+        || mt->IsNotTightlyPacked()
+        || mt->GetClass()->IsInlineArray())
     {
         mt->SetHasCheckedCanCompareBitsOrUseFastGetHashCode();
```

Comment:
This is an unfortunate consequence of throwing.

We'd otherwise be able to treat `InlineArray` of primitive types as bitwise equatable and allow them to participate in some of the vectorization speedups.

It would be nice if there was a better way to allow users to override the default while still opting their types into bitwise equality support.

Reply:
I think this is a point in time issue. This is merely the simplest way to block the scenario. It is entirely possible how we block this is changed in the future in a way that permits vectorization speed ups. Filed an issue: https://github.com/dotnet/runtime/issues/103712.

Reply:
👍, was more just calling it out as something for us to make explicitly known, not to block the change or anything like that.

It's just a broader issue where anytime the user overrides default behavior, we lose the ability to optimize things as we can no longer assume the implementation (even if the user implementation does exactly what we'd want).